AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_materialproviders/dafny_to_smithy.py

```diff
@@ -1210,6 +1231,15 @@ def aws_cryptography_materialproviders_CreateAwsKmsHierarchicalKeyringInput(
             if (dafny_input.cache.is_Some)
             else None
         ),
+        partition_id=(
+            (
+                b"".join(
+                    ord(c).to_bytes(2, "big") for c in dafny_input.partitionId.value
+                ).decode("utf-16-be")
```

Comment:
hmm i thought in the spec we say that the partition id is in utf8, is it expected for python to say `utf-16-be`?

Reply:
This is just a reflection of the Smithy model: https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/keyrings.smithy#L345-L346
Smithy model marks this as String, not DafnyUTF8Bytes.
Spec suggests the expected behavior is "input is provided as UTF-16/Dafny string, and implementation will convert and interpret it as UTF-8": https://github.com/awslabs/aws-encryption-sdk-specification/commit/c1bda2f9b6672b1825062cc625698a352ed4fb6e#diff-6a5370be266efa622d672e56f8c35f36dbc802d7bff63a1ea943338a115f0880R96-R101